src/guide/reactivity-computed-watchers.md

```diff
@@ -211,6 +213,37 @@ firstName.value = 'John' // logs: ["John", ""] ["", ""]
 lastName.value = 'Smith' // logs: ["John", "Smith"] ["John", ""]
 ```
 
+However, if you are changing both watched sources simultaneously in the same method, the watcher will be executed only once:
+
+```js{9-13}
+setup() {
```

Comment:
@skirtles-code the problem is example _works_. If we put the code from it directly to the `setup`, the logging will happen as it's shown in the example.

Sandbox to check: https://codesandbox.io/s/funny-firefly-gbnnx?file=/src/App.vue

Reply:
Why?  Why the example in your link, can be executed twice?

Reply:
[![fiujmD.png](https://z3.ax1x.com/2021/08/03/fiujmD.png)](https://imgtu.com/i/fiujmD)

Reply:
@zhenzhenChange it might be purely a CodeSandbox glitch when I think about it now 🤔 because it _should not_ work but it works.

<img width="598" alt="Screenshot 2021-08-03 at 09 55 52" src="https://user-images.githubusercontent.com/18719025/127979311-0781cf67-122a-4419-a1b1-0944b29b6d17.png">

Reply:
There are still some problems with this part, I need to test it, please merge and close it later

The version of the diagram I posted above is `Vue 3.1.4`

Here's an example from your link (`Vue 3.2.0.beta-7`): 
[![fiKTHg.png](https://z3.ax1x.com/2021/08/03/fiKTHg.png)](https://imgtu.com/i/fiKTHg)

Reply:
@zhenzhenChange tested locally with 3.1.4:
<img width="468" alt="Screenshot 2021-08-03 at 10 02 39" src="https://user-images.githubusercontent.com/18719025/127980227-cb7e0635-dbe6-45dd-9498-46d98e206c63.png">

It seems these two changes are considered to be two subsequent separate microtasks while joining them in a method breaks this behavior.

Reply:
I also stupid, this example at the time I was directly copy and paste to run in the local, you can see the upper right corner of the console in my figure, and `localhost:4321` it, and also tested with a few friends at the time.

Reply:
I'll start by writing a simple, multi-case example to test it, with a few conditions : 

- Vue version `3.1.4` or `latest`
- `<script>` or `<script setup>`
- `setTimeout` or `Promise`


I am also executing twice now, and I can't actually recover the scene of executing once 😱😱😱

Reply:
Heartbroken, just a few examples of simple testing, posted in the `issue` #1154

Reply:
@zhenzhenChange thank you for the detailed reproduction on the issue!

Reply:
> If we put the code from it directly to the `setup`, the logging will happen as it's shown in the example.

So it does. Did not expect that.

When I tried it myself I didn't put the code inside `setup`, I just ran it in top-level scope. Doing it that way doesn't log the first message.

I've explained what's going on here: https://github.com/vuejs/docs/issues/1154#issuecomment-894452072. It's a quirk of how watchers behave inside `setup` and, in my opinion, it should be changed in the core.

I'd also note that this problem isn't limited to watchers with multiple sources. A single source is sufficient to hit the problem, it just needs multiple changes to the watched value.

All that said, in the short term we're just going to have to document it as best we can...

Reply:
@skirtles-code I agree it should be changed in the core because this creates an ambiguity on watchers behaviour 👍🏻